### ISSUE TITLE
Complex Builder and JSON Performance. POJOs and Generics.

### ISSUE THREAD

**Issue by alvin-reyes** [authenteq/java-bigchaindb-driver#21]
Same notes as @bodia added. I saw some minor tweaks (e.g. the timeout configs could be moved to Globals or given as parameters) but considering this is WIP we will get there eventually.

On a side note I was thinking that the builders got a little too complicated maybe?
Also it seems to me that we're digging through JSONs a little bit too much and we should rely more on POJOs and maybe generics in the future?

**Comment by bodia** [comment]
Yeah, I have a feeling about complex builders as well. It's convenient, but also limits the flexibility to add certain changes
